src/components/navigation/Header.tsx

```diff
@@ -11,11 +11,44 @@ const gradientAnimation = `
 }
 `;
 
+function useScrollDirection() {
+  const [scrollDirection, setScrollDirection] = useState('up');
+  const [prevScroll, setPrevScroll] = useState(0);
```

Comment:
If we swap this to a ref we only get rerenders when it changes instead of every 10px scrolled